### DISCUSSION TITLE
C++ API documentation

### DISCUSSION THREAD

**Discussion by jeffj-archer** [OpenDDS/OpenDDS#4577] [Q&A]
Where can one find version/release specific OpenDDS C++ API documentation?  I've seen the "Developer's Guides" on OpenDDS's website, but those aren't API documents.

**Comment by iguessthislldo** [comment]
Unfortunately there isn't really one at the moment besides what's sprinkled into the Developer's Guide. We used to host a Doxygen of the latest major/minor release, but because of organizational circumstances we couldn't host it in the same place anymore (See: #4114). You should still be able to build it yourself using [`tools/scripts/generate_combined_doxygen.pl`](https://github.com/OpenDDS/OpenDDS/blob/master/tools/scripts/generate_combined_doxygen.pl). As I mentioned in the issue though, I don't think the Doxygen is particularly helpful, at least how it is right now. This is because it's including everything in OpenDDS, Doxygen comments are sparse in OpenDDS, and there's no integration between the comments and the Developer's Guide in the Sphinx documentation. So it's not much different from reading the source code in a IDE.

These [GitHub tasks](https://github.com/orgs/OpenDDS/projects/7) aim to solve these issues by integrating the Doxygen content into the Sphinx documentation and adding comprehensive stable API documentation. I'm planning on trying to start tackling that after the ongoing [configuration documentation improvments](https://github.com/OpenDDS/OpenDDS/pull/4467) are done for the upcoming release, but it's honestly it's probably going to be a big effort to document what needs to be documented.

**Comment by jeffj-archer** [comment]
Thanks for the info, much appreciated.
